wire-ios-sync-engine/Tests/Source/Calling/WireCallCenterV3Tests.swift

```diff
@@ -1807,7 +1846,81 @@ extension WireCallCenterV3Tests {
 
         // THEN
         XCTAssert(waitForAllGroupsToBeEmpty(withTimeout: 0.5))
+    }
 
+    func test_HandleActiveSpeakersChange_DoesntPostNotification_WhenActiveSpeakersChange_IsNotRelevant() throws {
+        // GIVEN
+        let conversationId = try XCTUnwrap(groupConversationID)
+
+        // We have three clients in the call
+        let clientOne = AVSClient.mockClient
+        let clientTwo = AVSClient.mockClient
+        let clientThree = AVSClient.mockClient
+
+        // We set the speaker levels for each client
+        let speakerOne = ActiveSpeaker(
+            userId: clientOne.userId,
+            clientId: clientOne.clientId,
+            audioLevel: 0,
+            audioLevelNow: 0
+        )
+
+        let speakerTwo = ActiveSpeaker(
+            userId: clientTwo.userId,
+            clientId: clientTwo.clientId,
+            audioLevel: 0,
+            audioLevelNow: 99
+        )
+
+        let speakerThree = ActiveSpeaker(
+            userId: clientThree.userId,
+            clientId: clientThree.clientId,
+            audioLevel: 0,
+            audioLevelNow: 38
+        )
+
+        // We create the call snapshot
+        let callSnapshot = CallSnapshotTestFixture.callSnapshot(
+            conversationId: conversationId,
+            callCenter: sut,
+            clients: [clientOne, clientTwo, clientThree],
+            activeSpeakers: [speakerOne, speakerTwo, speakerThree]
+        )
+
+        sut.callSnapshots = [conversationId: callSnapshot]
+
+        // We prepare the change of active speakers.
+        // They're not relevant if the audio level stays >0
+        let newSpeakerTwo = ActiveSpeaker(
+            userId: clientTwo.userId,
+            clientId: clientTwo.clientId,
+            audioLevel: 0,
+            audioLevelNow: 24
+        )
+
+        let newSpeakerThree = ActiveSpeaker(
+            userId: clientThree.userId,
+            clientId: clientThree.clientId,
+            audioLevel: 0,
+            audioLevelNow: 87
+        )
+
+        let change = AVSActiveSpeakersChange(activeSpeakers: [speakerOne, newSpeakerTwo, newSpeakerThree])
+
+        // We set the expectation for notifications
+        customExpectation(
+            forNotification: WireCallCenterActiveSpeakersNotification.notificationName,
+            object: nil
+        ) { _ in
```

Comment:
Maybe we can use [`XCTNSNotificationExpectation`](https://developer.apple.com/documentation/xctest/xctnsnotificationexpectation) instead of the custom one, that does a few opaque things?

Reply:
thanks for the suggestion, I wrongly assumed I needed to inherit from `XCTestCase` to be able to setup this notification expectation

Reply:
applied https://github.com/wireapp/wire-ios/pull/1507/commits/f8dfdca98ceb239bf997d7ff035b4e8bce051cb8